src/pages/IndividualReport.vue

```diff
@@ -0,0 +1,410 @@
+  enabled: initialized,
+  keepPreviousData: true,
+  staleTime: 5 * 60 * 1000,
+});
+
+const { data: administrationData } = useQuery({
+  queryKey: ['administrations', props.administrationId],
+  queryFn: () => fetchDocById('administrations', props.administrationId),
+  enabled: initialized,
+  keepPreviousData: true,
+  staleTime: 5 * 60 * 1000,
+});
+
+const expanded = ref(false);
+const exportLoading = ref(false);
+
+const setExpand = () => {
+  console.log('expand called');
+  expanded.value = !expanded.value;
+};
+
+const IndividualReportDownload = ref(null);
+
+const pageWidth = 190; // Set page width for calculations
+const returnScaleFactor = (width) => pageWidth / width; // Calculate the scale factor
+
+// Helper function to add an element to a document and perform page break logic
+const addElementToPdf = async (element, document, yCounter, offset = 0) => {
+  await html2canvas(element, { windowWidth: 1300, scale: 2 }).then(function (canvas) {
+    const imgData = canvas.toDataURL('image/jpeg', 0.7, { willReadFrequently: true });
+    const scaledCanvasHeight = canvas.height * returnScaleFactor(canvas.width);
+    // Add a new page for each task if there is no more space in the page for task desc and graph
+    if (yCounter + scaledCanvasHeight + offset > 287) {
+      document.addPage();
+      yCounter = 10;
+    } else {
+      // Add Margin
+      yCounter += 5;
+    }
+
+    document.addImage(imgData, 'JPEG', 10, yCounter, pageWidth, scaledCanvasHeight);
+    yCounter += scaledCanvasHeight;
+  });
+  return yCounter;
+};
+
+const exportToPdf = async () => {
+  exportLoading.value = true; // Set loading icon in button to prevent multiple clicks
+  if (!expanded.value) {
+    setExpand();
+  }
+  await new Promise((resolve) => setTimeout(resolve, 250));
+
+  const doc = new jsPDF();
+  let yCounter = 10; // yCounter tracks the y position in the PDF
+
+  // Add At a Glance Charts and report header to the PDF
+  const individualReport = document.getElementById('individual-report-export');
+  if (individualReport !== null) {
+    await addElementToPdf(individualReport, doc, yCounter);
+  }
+
+  doc.save(`IndividualScoreReport_${studentFirstName.value}${studentLastName.value}.pdf`),
+    (exportLoading.value = false);
+
+  return;
+};
+
+const tasks = computed(() => taskData?.value?.map((assignment) => assignment.taskId));
+
+const formattedTasks = computed(() => {
+  return (
+    tasks?.value
+      ?.map((taskId) => (extendedTaskTitle[taskId] ? extendedTaskTitle[taskId] : taskId))
+      .sort((a, b) => {
+        if (Object.keys(taskDisplayNames).includes(a.taskId) && Object.keys(taskDisplayNames).includes(b.taskId)) {
+          return taskDisplayNames[a.taskId].order - taskDisplayNames[b.taskId].order;
+        } else {
+          return -1;
+        }
+      })
+      .join(', ') + '.'
+  );
+});
+
+const studentFirstName = computed(() => {
+  if (studentData.value.name.first == undefined) return studentData.value.username;
```

Comment:
This might need conditional chaining
```suggestion
  if (studentData.value.name?.first == undefined) return studentData.value.username;
```